### PR TITLE
Create Name model for usernames

### PR DESCRIPTION
**Pull request by jace**
Merges name from User and Organization models into a single table, thereby fulfilling the unification requirement described in #91 and #232.

### CODE REVIEW

**Comment by jace** [comment]
This change also fixes username validation in the models, in the process exposing some bugs in the unit tests. These have now been corrected.

File: lastuser_core/models/user.py [hunk 1]
```diff
@@ -5,37 +5,116 @@
...
+        return cls.query.filter(db.func.lower(Name.name) == db.func.lower(name)).one_or_none()
+
+    @db.validates('name')
+    def validate_name(self, key, value):
+        if not valid_username(value):
+            raise ValueError("Invalid username: " + value)
+        return value
+
 
+class SharedNameMixin(object):
```
Comment (iambibhas) [inline lastuser_core/models/user.py]: What is this model for?

File: lastuser_oauth/forms/profile.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def validate_old_password(self, field):
 
 class ProfileForm(forms.Form):
     fullname = forms.StringField(__("Full name"),
-        validators=[forms.validators.DataRequired(), forms.validators.Length(max=80)])
+        validators=[forms.validators.DataRequired(), forms.validators.Length(max=63)])
```
Comment (iambibhas) [inline lastuser_oauth/forms/profile.py]: Why the length restriction on full name? Username is created from it?

File: lastuser_core/models/user.py (hunk 1 above)
Comment (jace) [reply lastuser_core/models/user.py]: It's not a model. It offers the `name` property that makes the relationship behave like a string in both `User` and `Organization`.

File: lastuser_oauth/forms/profile.py (hunk 2 above)
Comment (jace) [reply lastuser_oauth/forms/profile.py]: DNS labels are restricted to 63 characters. Any longer and Talkfunnel's subdomains will break.

Comment (iambibhas) [reply lastuser_oauth/forms/profile.py]: No I know that and get the limit for username. full name is never used on url, right?

Comment (jace) [reply lastuser_oauth/forms/profile.py]: Whoops, validator needed to be on the other field.

**Comment by jace** [comment]
After having tried the custom comparator approach, I've found that there is no way to invisibly insert a join condition. Whoever is performing the query must (a) perform the join, or (b) [call `with_transformation` as described in the documentation](https://docs.sqlalchemy.org/en/latest/orm/extensions/hybrid.html#building-transformers). Since our problem  is that organization views use load_models, which isn't aware of these things, the best approach seems to be to deprecate load_models and use ClassView with a custom loader.

I'm therefore merging this PR as is, and will pursue ClassView in a separate PR.

**Pull request merged**